src/main/java/org/folio/services/voucher/BatchVoucherGenerateService.java

```diff
@@ -72,17 +72,17 @@ public Future<BatchVoucher> buildBatchVoucherObject(BatchVoucherExport batchVouc
     String voucherCQL = buildBatchVoucherQuery(batchVoucherExport);
     return voucherService.getVouchers(voucherCQL, 0, Integer.MAX_VALUE, requestContext)
       .compose(vouchers -> {
-        if (!vouchers.getVouchers().isEmpty()) {
-          Future<Map<String, List<VoucherLine>>> voucherLines = voucherLineService.getVoucherLinesMap(vouchers, requestContext)
-            .onFailure(t -> log.error("buildBatchVoucherObject:: Error retrieving voucher lines", t));
-          Future<Map<String, Invoice>> invoices = invoiceRetrieveService.getInvoiceMap(vouchers, requestContext)
-            .onFailure(t -> log.error("buildBatchVoucherObject:: Error retrieving invoices", t));
-          Future<Map<String, List<InvoiceLine>>> invoiceLines = invoiceLinesRetrieveService.getInvoiceLineMap(vouchers, requestContext)
-            .onFailure(t -> log.error("buildBatchVoucherObject:: Error retrieving invoice lines", t));
-          return CompositeFuture.join(voucherLines, invoices, invoiceLines)
-            .compose(v -> buildBatchVoucher(batchVoucherExport, vouchers, voucherLines.result(), invoices.result(), invoiceLines.result(), requestContext));
+        if (vouchers.getVouchers().isEmpty()) {
+          throw new HttpException(404, "Vouchers for batch voucher export were not found");
         }
-       throw new HttpException(404, "Vouchers for batch voucher export were not found");
+        Future<Map<String, List<VoucherLine>>> voucherLines = voucherLineService.getVoucherLinesMap(vouchers, requestContext)
+          .onFailure(t -> logger.error("buildBatchVoucherObject:: Error retrieving voucher lines", t));
+        Future<Map<String, Invoice>> invoices = invoiceRetrieveService.getInvoiceMap(vouchers, requestContext)
+          .onFailure(t -> logger.error("buildBatchVoucherObject:: Error retrieving invoices", t));
+        Future<Map<String, List<InvoiceLine>>> invoiceLines = invoiceLinesRetrieveService.getInvoiceLineMap(vouchers, requestContext)
+          .onFailure(t -> logger.error("buildBatchVoucherObject:: Error retrieving invoice lines", t));
```

Comment:
please use this pattern in other places as well

Reply:
one thing I forgot, we don't need to put method name to log.error it is common patter used in many places including logging improvement documentation

Reply:
We don't need it when we have exception and stack trace but what about when we are logging just message?

Reply:
maybe method name is useful only log.debug, warn, info cases. error cases has mostly error stacktrace so that we don't need to write method name